pkg/tfshim/shim.go

```diff
@@ -42,6 +42,9 @@ type InstanceDiff interface {
 	Destroy() bool
 	RequiresNew() bool
 
+	// Deprecated. Instead of constructing an InstanceDiff object and then calling IgnoreChanges
+	// on it, please pass IgnoreChanges option to Diff so that the ignores are processed as part
+	// of constructing InstanceDiff.
 	IgnoreChanges(ignored map[string]bool)
```

Comment:
Is there a reason we are not just removing This? We are already making a breaking change to all consumers of `shim/*` (changing the method signature of `Diff` and `DiffWithContext`.

---

If we decide to keep, let's use `:` instead of `.`, since that triggers the `staticcheck` linter.

```suggestion
	// Deprecated: Instead of constructing an InstanceDiff object and then calling IgnoreChanges
	// on it, please pass IgnoreChanges option to Diff so that the ignores are processed as part
	// of constructing InstanceDiff.
	IgnoreChanges(ignored map[string]bool)
```

Reply:
Well adding functional options doesn't break callers just implementers but yeah. I'm mostly thinking of not causing issues for Cosmic but perhaps that's a moot concern. If we can take breaking changes over these internals that's good.